### DISCUSSION TITLE
Clangd can changed cache file from rename request?

### DISCUSSION THREAD

**Discussion by Hebe970217** [clangd/clangd#1050] [Q&A]
I watch lsp protocol, the clangd is not support file notify operation from itself, if my IDE front changed edit file, that now sync to clangd, need to use didChange request, is right?

**Comment by sam-mccall** [comment]
For parsing a file, clangd takes that file's content from the last didChange notification, and all headers it includes are read from disk. (Even if those headers are also open with unsaved changes).

didChange triggers reparsing of the changed file. didSave triggers reparsing of all files (in case the saved file was a header included by open files).

clangd does not watch for filesystem change events either through LSP or on its own.
Changes to on-disk content of included headers will be picked up on the next reparse.

**Reply by Hebe970217** [reply]
If according to the cross platform design, the user passes text to Clangd through didchange request, it will not open from disk, right?

**Reply by Hebe970217** [reply]
The LSP protocol itself is cross platform. I can frontend on X86 machines redirect the backend to other platforms . Does Clangd itself strongly depend on the local file system?
